workers/unity/Packages/com.improbable.gdk.deploymentlauncher/DeploymentLauncher.cs

```diff
@@ -0,0 +1,334 @@
+using Fps;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Tools;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEditor;
+using UnityEngine;
+
+namespace Improbable.Gdk.DeploymentManager
+{
+    internal class DeploymentManager
+    {
+        private const string DeploymentLauncherMenuItem = "SpatialOS/Deployment Launcher";
+        private const int DeploymentLauncherPriority = 51;
+
+        private static readonly string DotNetProjectPath =
+            Path.GetFullPath(Path.Combine(Tools.Common.GetPackagePath("com.improbable.gdk.deploymentlauncher"), ".DeploymentLauncher/DeploymentLauncher.csproj"));
+        private static readonly string DotNetWorkingDirectory = Path.GetFullPath(Path.Combine(Application.dataPath, ".."));
+        private static readonly string ProjectRootPath =
+            Path.Combine(Application.dataPath, "../../../");
+
+        [MenuItem(DeploymentLauncherMenuItem, false, DeploymentLauncherPriority)]
+        private static void LaunchDeploymentMenu()
+        {
+            // Show existing window instance. If one doesn't exist, make one.
+            var inspectorWindowType = typeof(EditorWindow).Assembly.GetType("UnityEditor.InspectorWindow");
+            var deploymentWindow = EditorWindow.GetWindow<DeploymentEditorWindow>(new Type[] { inspectorWindowType });
+            deploymentWindow.titleContent.text = "Deployments";
+            deploymentWindow.titleContent.tooltip = "A tab for managing your SpatialOS deployments.";
+            deploymentWindow.Show();
+        }
+
+        internal class DeploymentInfo
+        {
+            public string Name;
+            public string Id;
+        }
+
+        internal class DeploymentEditorWindow : EditorWindow
+        {
+            private string projectName = "unity_gdk";
+            private string assemblyName;
+            private string deploymentName;
+            private string snapshotPath = "snapshots/cloud.snapshot";
+            private string mainLaunchJson = "cloud_launch_small.json";
+            private string simWorkerLaunchJson = "cloud_launch_small_sim_workers.json";
+            private string simWorkerDeploymentName;
+            private bool simWorkerDeploymentEnabled;
+            private bool simWorkerCustomDeploymentNameEnabled;
+
+            List<DeploymentInfo> deploymentList;
+            int selectedDeployment;
+
+            private Task<bool> runningLaunchTask;
+            private Task<bool> runningStopTask;
+            private Task<List<DeploymentInfo>> runningListTask;
+
+            void OnEnable()
+            {
+                deploymentList = new List<DeploymentInfo>();
+            }
+
+            void OnGUI()
```

Comment:
General comment for this PR:

I definitely recommend looking to `EditorGUI.DisabledScope`, and `EditorGUILayout.HorizontalScope`, et al.
Examples:
https://docs.unity3d.com/ScriptReference/EditorGUI.DisabledScope.html
https://docs.unity3d.com/ScriptReference/EditorGUILayout.ToggleGroupScope.html

They really do make the code more readable and you don't have to worry about mismatching Begin/End pairs when refactoring (it really does become a maintenance pain)

Reply:
Done. Much nicer 👌